### PR TITLE
Upgrade libebur128 from 1.2.4-patched to 1.2.6

### PR DESCRIPTION
**Pull request by uklotzde**
They released a new version after over 2 years!! Perfectly in time for our 2.3.0 release.

Updated our bundled, intermediate version 1.2.4+ to the current, official release. Version 1.2.5 is sufficient for all other platforms except Windows and should be provided by most up-to-date Linux distributions.

~~I applied the same .gitignore tweak that we used for googletest after deleting the unneeded files in the first step.~~ The test directory is added unconditionally even if tests are disabled.

### CODE REVIEW

**Comment by uklotzde** [comment]
> macOS build fails:
> 
> ```
>   add_subdirectory given source "test" which is not an existing directory.
> ```
> 
> Maybe we can't remove the tests.

Non-sophisticated build file. Fixed.

**Comment by Holzhaus** [comment]
What was the reason for the custom patches. Are you sure they don't apply anymore?

**Comment by uklotzde** [comment]
> What was the reason for the custom patches. Are you sure they don't apply anymore?

The Windows build failed for the official 1.2.4 release due to SSE stuff, I don't remember exactly. We had to cherry-pick one or more commits to fix those issues.

**Comment by uklotzde** [comment]
Now the macOS build doesn't find the library. WTF!?

**Comment by uklotzde** [comment]
Oh, they have renamed the build options :see_no_evil:

**Comment by JoergAtGithub** [comment]
> 
> 
> > What was the reason for the custom patches. Are you sure they don't apply anymore?
> 
> The Windows build failed for the official 1.2.4 release due to SSE stuff, I don't remember exactly. We had to cherry-pick one or more commits to fix those issues.

The Windows builds are not using 1.24patched anymore, because the Windows buildenv contains 1.26 already.

**Comment by daschuer** [comment]
Does this fix anything? Do we want to take the risk of a risk minute merge? This seems to be not a no-brainer.

File: cmake/modules/FindEbur128.cmake [hunk 1]
```diff
@@ -45,7 +45,8 @@ The following cache variables may also be set:
 
 find_package(PkgConfig QUIET)
 if(PkgConfig_FOUND)
-  pkg_check_modules(PC_Ebur128 QUIET libebur128>=1.2.4)
+  # The bundled version 1.2.6 only fixes dynamic linking on Windows
+  pkg_check_modules(PC_Ebur128 QUIET libebur128>=1.2.5)
```
Comment (daschuer) [inline cmake/modules/FindEbur128.cmake:49]: Is the version check correct here? For my understanding it belongs to the CMakeList.txt file.

**Comment by uklotzde** [comment]
> Does this fix anything? Do we want to take the risk of a risk minute merge? This seems to be not a no-brainer.

Why bundle an outdated version if the majority of installations already uses an updated version???

**Comment by Holzhaus** [comment]
I think this is safe, because a) it's just a Bugfix release and b) all windows builds already use it anyway, as well as most people on Linux probably.

**Comment by daschuer** [comment]
Because it works without any issues. I think we should consider to remove it, Ubuntu Bionic has also 1.2.4-1. 
But that is IMHO not a last minute task.

**Comment by daschuer** [comment]
All this is legacy to support Ubuntu Trusty 14.4 
https://github.com/mixxxdj/mixxx/pull/520

**Comment by daschuer** [comment]
> In general I think it's better to use the same version on different OSes.

This argument applies to all libraries and does not match to the philosophy of non rolling distros.

File: cmake/modules/FindEbur128.cmake (hunk 1 above)
Comment (uklotzde) [reply cmake/modules/FindEbur128.cmake:49]: This would be an unrelated change of the build. Out of scope here.

**Comment by uklotzde** [comment]
We are already shipping an unofficial version in 2.2 which is bad practice.

If this version causes issues we would need to prevent all dynamic linking and roll back the Windows build environment.

**Comment by uklotzde** [comment]
@daschuer Even @Be-ing has helped to get this release ready for vcpkg/Windows, especially for Mixxx 2.3! And now you refuse to use it for all of our platforms?

**Comment by daschuer** [comment]
I am not refusing anything.

I have just expressed my concerns that we change 2.3 without a strong need. 

The underlying issue is that I want to get rid of this libray from our lib folders, because the original  reason to support Trusty is no longer valid.

For MacOs I thing the best way forward would be to install the library in the build environment.

Currently all Linux distros, we are supporting, are useing their native library version. The upgrade will affect all active Ubuntu versions. 

Not using the system library should be an exception for a strong reason. 

After we merge this upgrade, ripping off the library will effectively downgrade the version and can be considered as regression. So this merge will make it hard to do the step in the next two years until Ubuntu Impish 21.10 falls out of support. 

So let's ask ourselves if the reason is strong enough for the upgrade. If you considered yes, I have no objections to merge this.

**Pull request closed without merging**